### PR TITLE
Add setsid implementation

### PR DESCRIPTION
**Pull request by kov**
This one is very straight-forward. I did both an implementation using libc to fork and exec directly and one using the std::process::Command interface. I am not sure how much is gained in terms of clarity or safety, given how straight-forward this tool is, let me know which one you prefer =)

The main open questions I have are 1) whether we should first get the handling of the error codes landed in uucore:

```rust
    // FIXME: POSIX wants certain exit statuses for specific errors, should
    // these be handled by uucore::error? We should be able to just return
    // the UError here.
    uucore::show_error!(
        "failed to execute {}: {}",
        executable.to_string_lossy(),
        UIoError::from(error)
    );

    if kind == io::ErrorKind::NotFound {
        uucore::error::set_exit_code(127);
    } else if kind == io::ErrorKind::PermissionDenied {
        uucore::error::set_exit_code(126);
    }
```

2) whether it makes sense to put some effort into extracting the testing infrastructure out of coreutils (or making it usable as a dependency, akin to uucore), instead of carrying a separate copy here, so we don't need to keep cherry-picking features and fixes.

### CODE REVIEW

File: src/uu/setsid/setsid.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# setsid
+
+```
+setsid [options] <program> [argument ...]
+```
+
+Run a command as the leader of a new session.
```
Comment (sylvestre) [inline src/uu/setsid/setsid.md:7]:
```suggestion
Run a program in a new session.
```

File: src/uu/setsid/src/setsid.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
+use clap::builder::ValueParser;
```
Comment (sylvestre) [inline src/uu/setsid/src/setsid.rs:1]: please add the license header

**Comment by kov** [comment]
Getting there for windows, I'll do some testing on macos locally to understand how it should work before I submit the next one, which should finish clearing up windows as well.

I expect we do want to support macos if possible, right?

**Comment by sylvestre** [comment]
yeah, except if you have excellent reasons not to (ex: not a thing on mac)

**Comment by kov** [comment]
@sylvestre what do you think on the matter of raw libc (first commit) vs Command (second), do you have a preference?

**Comment by kov** [comment]
Oh, and do those error codes for not found / permission denied being handled by uucore::error sounds like the appropriate solution to you, or do you prefer keeping that on the utility side?

File: src/uu/setsid/src/setsid.rs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+#[cfg(target_family = "unix")]
+fn report_failure_to_exec(error: io::Error, executable: &OsStr, set_error: bool) {
+    let kind = error.kind();
+
+    // This error comes from the ioctl we run inside pre_exec().
+    if kind == io::ErrorKind::Other {
+        uucore::error::set_exit_code(1);
+        uucore::show_error!("{}", error.to_string());
```
Comment (sylvestre) [inline src/uu/setsid/src/setsid.rs:41]: it is possible to trigger this error in the tests?

File: src/uu/setsid/src/setsid.rs [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        }
+    }
+}
+
+#[cfg(not(target_family = "unix"))]
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let _matches: clap::ArgMatches = uu_app().try_get_matches_from(args)?;
+
+    Err(uucore::error::USimpleError::new(
```
Comment (sylvestre) [inline src/uu/setsid/src/setsid.rs:68]: please add a windows test to see this error

File: src/uu/setsid/src/setsid.rs [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches: clap::ArgMatches = uu_app().try_get_matches_from(args)?;
+
+    let force_fork = matches.get_flag("fork");
+    let wait_child = matches.get_flag("wait");
+    let take_controlling_tty = matches.get_flag("ctty");
+
+    let command: Vec<_> = match matches.get_many::<OsString>("command") {
+        Some(v) => v.collect(),
+        None => return Err(uucore::error::USimpleError::new(1, "no command specified")),
```
Comment (sylvestre) [inline src/uu/setsid/src/setsid.rs:85]: please add a test to fall into this error

File: src/uu/setsid/src/setsid.rs [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                );
+                libc::_exit(1);
+            }
+
+            Ok(())
+        });
+    };
+
+    if force_fork || already_group_leader() {
+        let mut child = match to_run.spawn() {
```
Comment (sylvestre) [inline src/uu/setsid/src/setsid.rs:123]: maybe move this block into a function

File: src/uu/setsid/src/setsid.rs [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+    // We know we have at least one item, as none was
+    // handled as an error on the match above.
+    let executable = command[0];
+    let arguments = command.get(1..).unwrap_or(&[]);
+
+    let mut to_run = process::Command::new(executable);
+    to_run.args(arguments.iter());
+
+    unsafe {
+        to_run.pre_exec(move || {
```
Comment (sylvestre) [inline src/uu/setsid/src/setsid.rs:97]: maybe move this block into a function

File: src/uu/setsid/src/setsid.rs [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+    // We know we have at least one item, as none was
+    // handled as an error on the match above.
+    let executable = command[0];
+    let arguments = command.get(1..).unwrap_or(&[]);
+
+    let mut to_run = process::Command::new(executable);
+    to_run.args(arguments.iter());
+
+    unsafe {
```
Comment (sylvestre) [inline src/uu/setsid/src/setsid.rs:96]: all the functions needs to be in an unsafe block ?
please document it

**Pull request merged**